Feature/ability to select outgoing number

vnagalingam opened the pull request:
Added new field to show the outgoing from number
This number is will be atwilio number associated with the agent's skill. If the agent has multiple skills, all the numbers associated with the skills will be shown in the outgoing from number field as a dropdown. 
If the agent has no skill, the default flex outbound dial number will be used as from number. This will not be displayed on the dialpad.

ishvar-kanajariya committed:
AGEUK-254 Need the ability to define the Outgoing number when making outbound calls

ishvar-kanajariya committed:
- UI changes

ishvar-kanajariya committed:
- Handle empty api response

danhyrjak commented on src/customActions/index.js:
```diff
@@ -103,4 +103,11 @@ export default (manager) => {
+  Actions.replaceAction("StartOutboundCall", (payload, original) => {
+    const state = manager.store.getState();
+    var newPayload = payload;
+    newPayload.callerId = state.dialpad.outgoingNumber.selectedOutgoingNumber?.phoneNumber;
```
Need to clear the value once you have used it, as discussed, you only want to use this number once. Note: from reading through the code I don't think this currently happens but might be worth a quick test first just to make sure I didn't miss anything :-).

vnagalingam committed:
fixed bug - do not outgoing number only where the agent has no skill ( i.e no phone numbers returned)
Removed isFetched checking in Component mount, agents skills may change at any time, so the phone numbers should be loaded each time the component is mounted.
